core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -13321,6 +13323,284 @@ class KafkaApisTest extends Logging {
+        .setPartitionLogInfo(Collections.singletonList(new GetReplicaLogInfoResponseData.PartitionLogInfo()
+          .setErrorCode(err.code())
+          .setPartition(1))))
+      idx += 1
+    }
+
+
+    val builder = new GetReplicaLogInfoRequest.Builder(
+      new GetReplicaLogInfoRequestData().setTopicPartitions(tps asJava)
+    )
+    verifyGetReplicaLogInfoRequest(builder, { response =>
+      assertEquals(expected, response.data())
+    })
+  }
+
+  @Test
+  def testGetReplicaLogInfoFailedToRetrieveLog(): Unit = {
+    val topic1 = new GetReplicaLogInfoRequestData.TopicPartitions()
+      .setTopicId(Uuid.randomUuid())
+      .setPartitions(Collections.singletonList(1))
+    val topic2 = new GetReplicaLogInfoRequestData.TopicPartitions()
+      .setTopicId(Uuid.randomUuid())
+      .setPartitions(Collections.singletonList(2))
+    val builder = new GetReplicaLogInfoRequest.Builder(List(topic1, topic2) asJava)
+
+    metadataCache = mock(classOf[MetadataCache])
+    when(metadataCache.getTopicName(topic1.topicId())).thenReturn(Optional.of("topic1"))
+    when(metadataCache.getTopicName(topic2.topicId())).thenReturn(Optional.of("topic2"))
+
+    val log1 = mock(classOf[UnifiedLog])
+    when(log1.logEndOffset).thenReturn(100L)
+    when(log1.latestEpoch).thenReturn(Optional.of(10))
+    val partition1 = mock(classOf[Partition])
+    when(partition1.log).thenReturn(Some(log1))
+    when(partition1.getLeaderEpoch).thenReturn(1)
+    when(partition1.partitionId).thenReturn(1)
+
+    val partition2 = mock(classOf[Partition])
+    when(partition2.log).thenReturn(None)
+    when(partition2.getLeaderEpoch).thenReturn(2)
+    when(partition2.partitionId).thenReturn(2)
+
+    val tp1 = new TopicPartition("topic1", 1)
+    when(replicaManager.getPartitionOrError(tp1)).thenReturn(Right(partition1))
+    val tp2 = new TopicPartition("topic2", 2)
+    when(replicaManager.getPartitionOrError(tp2)).thenReturn(Right(partition2))
+
+    val expected = new GetReplicaLogInfoResponseData()
+      .setTopicPartitionLogInfoList(List(
+        new GetReplicaLogInfoResponseData.TopicPartitionLogInfo()
+          .setTopicId(topic1.topicId())
+          .setPartitionLogInfo(Collections.singletonList(
+            new GetReplicaLogInfoResponseData.PartitionLogInfo()
+              .setPartition(1)
+              .setLogEndOffset(100L)
+              .setLastWrittenLeaderEpoch(10)
+              .setCurrentLeaderEpoch(1))),
+        new GetReplicaLogInfoResponseData.TopicPartitionLogInfo()
+          .setTopicId(topic2.topicId())
+          .setPartitionLogInfo(Collections.singletonList(
+            new GetReplicaLogInfoResponseData.PartitionLogInfo()
+              .setErrorCode(Errors.LOG_DIR_NOT_FOUND.code())
+          ))) asJava)
+      .setBrokerEpoch(brokerEpoch)
+
+    verifyGetReplicaLogInfoRequest(builder, { response =>
+      assertEquals(expected, response.data())
+    })
+  }
+
+  @Test
+  def testGetReplicaLogInfoUnknownTopic(): Unit = {
```

Comment:
may be worth having one of these error-case tests confirm that requests with a mix of "good" and "bad" topics/partitions will not impact the return of information of the "good" partition